lib/Parse/ParseStmt.cpp

```diff
@@ -840,21 +840,14 @@ ParserResult<Stmt> Parser::parseStmtYield(SourceLoc tryLoc) {
 
     SyntaxParsingContext YieldsCtxt(SyntaxContext, SyntaxKind::YieldList);
 
-    SmallVector<Identifier, 4> yieldLabels;
-    SmallVector<SourceLoc, 4> yieldLabelLocs;
-    SmallVector<TrailingClosure, 2> trailingClosures;
-
-    status = parseExprList(tok::l_paren, tok::r_paren,
-                           /*postfix (allow trailing closure)*/ false,
-                           /*expr basic (irrelevant)*/ true,
-                           lpLoc,
-                           yields, yieldLabels, yieldLabelLocs,
-                           rpLoc,
-                           trailingClosures,
-                           SyntaxKind::ExprList);
-    assert(trailingClosures.empty());
-    assert(yieldLabels.empty());
-    assert(yieldLabelLocs.empty());
+    SmallVector<ExprListElt, 4> yieldElts;
+    status = parseExprList(tok::l_paren, tok::r_paren, /*isArgumentList*/ false,
+                           lpLoc, yieldElts, rpLoc, SyntaxKind::ExprList);
+    for (auto &elt : yieldElts) {
+      assert(elt.Label.empty());
+      assert(elt.LabelLoc.isInvalid());
```

Comment:
(Unrelated to this PR)
Hmm, I just noticed the following code hits these assertion even with current `main`.
```swift
struct S {
  var property: String {
    _read {
      yield(x: "test")
    }
  }
}
```
Instead of skipping parsing labels in `parseExprList()`, we probably should diagnose parsed labels here. Or, we should not use `parseExprList`.

Reply:
Good catch! Looks like the logic in `parseList` is trying to handle this case, but isn't working as `SyntaxKind::ExprList` is being passed (though looking at the git history, this seems to have been changed to fix the syntax tree). I can take a look at this in a follow-up.